src/iuse.cpp

```diff
@@ -4683,11 +4683,6 @@ int iuse::set_trap(player *p, item *it, bool, point)
         return 0;
     }
 
-    if (dirx == p->posx() && diry == p->posy()) {
```

Comment:
We need a trap for is_benign. At this point in the code we only have an item. The big block of conditions below this is where the item gets mapped to a trap type.